src/files/DCDFile.cpp

```
@@ -0,0 +1,457 @@
+   {
+      freeAtoms[atmIdx] = freeAtomsIndexes[atmIdx];
+   }
+}
+
+bool DCDFile::hasPositions() const
+{
+   if (getNumAtoms() != 0)
+   {
+      return true;
+   }
+
+   return false;
+}
+
+int DCDFile::getNumTitles() {
+   return NTITLE;
+}
+
+void DCDFile::setNumTitles(int titlesNumber) {
+   NTITLE = titlesNumber;
+}
+
+char* DCDFile::getTitle()
+{
+   return TITLE;
+}
+
+void DCDFile::setTitle(std::string title)
+{
+   TITLE = new char[NTITLE * 80 + 1];
+   char* allTitle = const_cast<char*>(title.c_str());
+   for (size_t ch = 0; ch < title.size(); ch++)
+   {
+      TITLE[ch] = allTitle[ch];
+   }
+   TITLE[NTITLE * 80] = '\0';
+}
+
+/*
+ * When writing a block of binary data to a file, Fortran adds 2 unsigned integer before and after the block : each one contains the total number of bytes of the block.
+ *
+ * For example for a fortran array of real(8) of size 10, 80 bytes of a data have to be written : so 3 things are written to the binary file :
+ *  1) An unsigned integer (4 bytes) , its value is 80
+ *  2) The fortran array (80 bytes)
+ *  3) A second unsigned integer (4 bytes), same value of 80.
+ * This corresponds to the fortran statement "write(file_descriptor) fortran_array"
+ *
+ * Now when writing several things at the same tine, i.e. "write(file_descriptor) fortran_array,an_integer"
+ * things are written as following:
+ * 1) unsigned int size
+ * 2) fortran array
+ * 3) an_integer
+ * 4) unsigned int size
+ * The value of the 2 unsigned ints is 84 : 80 for the array, 4 for the integer(4).
+ *
+ * The following method checkFortranIOerror check that the 2 unsigned ints have the same value: if not there was a probem when reading the binary fortran file.
+ */
+bool DCDFile::checkFortranIOerror(const char file[], const int line, const size_t fortcheck1, const size_t fortcheck2) const
+{
+   if (fortcheck1 != fortcheck2)
+   {
+      cout << "Error when reading data from dcd : quantities do not match." << endl;
```

Please throw an exception instead of writing to cout.